distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java

```diff
@@ -1424,54 +1447,244 @@ public void testProductDownloadListEmptyWhenNoDownloads() {
+        String startDownloadUrlFail = setUpDownloadResponse(failMetacardId, failFilename, failResourceData, FAIL_RETRIES, delayBetweenRetries);
+
+        String restUrlFail = REST_PATH.getUrl() + "sources/" + CSW_STUB_SOURCE_ID + "/" + failMetacardId
+                + "?transform=resource" + "&session=" + cometDClient.getClientId();
+
+        // Verify that product retrieval fails from the csw stub server.
+        // @formatter:off
+        when().get(restUrlFail).then().log().all().assertThat().statusCode(500).contentType("text/plain")
+                .body(containsString("cannot retrieve product"));
+        // @formatter:on
+
+        String restUrl = REST_PATH.getUrl() + "sources/" + CSW_STUB_SOURCE_ID + "/" + metacardId1
+                + "?transform=resource" + "&session=" + cometDClient.getClientId();
+
+        // Verify that the testData from the csw stub server is returned.
+        // @formatter:off
+        when().get(restUrl).then().log().all().assertThat().contentType("text/plain")
+                .body(is(resourceData1));
+        // @formatter:on
+
+        expect("Waiting for notifications").within(10, SECONDS)
+                .until(() -> foundExpectedNotifications(cometDClient, filename1, failFilename, NOTIFICATION_COMPLETE_MESSAGE, NOTIFICATION_FAILED_MESSAGE));
+        expect("Waiting for activities").within(10, SECONDS)
+                .until(() -> foundExpectedActivities(cometDClient, filename1, failFilename, ACTIVITES_STARTED_MESSAGE, ACTIVITES_STARTED_MESSAGE));
+
+        // @formatter:off
+        String downloadId1 = when().get(startDownloadUrl1).then().log().all()
+                .extract().jsonPath().getString("downloadId");
+        // @formatter:on
+        assertThat(downloadId1, not(isEmptyString()));
+
+        // @formatter:off
+        String failDownloadId = when().get(startDownloadUrlFail).then().log().all()
+                .extract().jsonPath().getString("downloadId");
+        // @formatter:on
+        assertThat(downloadId1, not(isEmptyString()));
+
+    }
+
+    /**
+     * Helper method used to determine that certain messages are showing up in the cometDClient notifications.
+     * @param cometDClient the CometDClient to check for messages.
+     * @param filename1 the filename of the first resource to check against the CometDClient.
+     * @param filename2 the filename of the second resource to check against the CometDClient.
+     * @param messageToFind1 the message to find in the CometDClient notifications related to the first filename.
+     * @param messageToFind2 the message to find in the CometDClient notifications related to the second filename.
+     * @return a boolean that is only true when both messages have been found in the notifications.
+     */
+    private boolean foundExpectedNotifications(CometDClient cometDClient, String filename1, String filename2, String messageToFind1, String messageToFind2)
```

Comment:
And to keep this simple, maybe the method should only take one message, one filename and throw an `IllegalArgumentException` if any of those is empty. I think it's ok if we need to call it twice in a test since we're not expecting to have to look at hundreds of messages.